### PR TITLE
Fix the teleport issue for BM

### PR DESCRIPTION
**Pull request by iman512003**
This is to address #405 .

### CODE REVIEW

File: smarts/core/smarts.py [hunk 1]
```diff
@@ -449,6 +449,7 @@ def teardown(self):
 
     def destroy(self):
         self.teardown()
+        self._traffic_sim.destroy()
```
Comment (iman512003) [inline smarts/core/smarts.py:452]: I am not sure whether this is the best place to add this function.

File: smarts/core/sumo_traffic_simulation.py [hunk 2]
```diff
@@ -108,11 +106,10 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
-    def _destroy(self):
-        if atexit:
-            atexit.unregister(self._destroy)
-
+    def destroy(self):
         self._close_traci_and_pipes()
+        if not self._is_setup:
```
Comment (iman512003) [inline smarts/core/sumo_traffic_simulation.py:111]: If I do not add this then the traffic test will fail when it calls for destroying smarts and throw an error which mentioned that no sumo process exists.

File: smarts/core/smarts.py (hunk 1 above)
Comment (Gamenot) [reply smarts/core/smarts.py:452]: This should be fine, although, it might be better to put it after the `self._agent_manager.destroy()`

**Review by Gamenot — APPROVED** [review]
LGTM

**Review by Adaickalavan — APPROVED** [review]

File: smarts/core/smarts.py (hunk 1 above)
Comment (liamchzh) [reply smarts/core/smarts.py:452]: Should be fine.

**Review by liamchzh — APPROVED** [review]

**Pull request merged**